### ISSUE TITLE
Bad opitmization for MergePadConvOptimizer

### ISSUE THREAD

**Issue by edmBernard** [microsoft/onnxconverter-common#176]
I have a maybe strange architecture that pipe two padding in the same convolution. 
A small example can be create with keras using the simple code.
    
    def padding_network_v2(input_layer, nf=64):
        temp = keras.layers.Conv2D(filters=nf, kernel_size=(3, 3), strides=(1, 1), padding="same")(input_layer)
        pad1 = keras.layers.ZeroPadding2D(padding=((1, 0), (1, 0)))(temp)
        pad2 = keras.layers.ZeroPadding2D(padding=((0, 1), (0, 1)))(temp)
        conv = keras.layers.Conv2D(filters=nf, kernel_size=(3, 3), strides=(1, 1), padding="valid")
        output = keras.layers.concatenate([conv(pad1), conv(pad2)], axis=3)
    
        return output
        
![image](https://user-images.githubusercontent.com/19322406/113677025-66a4e500-96bd-11eb-9eb2-b4e321597a9e.png)

During optimization the function MergePadConvOptimizer optimize one of the convolution and accidentally remove the second.

The optimization code work correctly when padding operation just follow the input nodes for example : 

    def padding_network(input_layer, nf=64):
        pad1 = keras.layers.ZeroPadding2D(padding=((1,0), (1,0)))(input_layer)
        pad2 = keras.layers.ZeroPadding2D(padding=((0,1), (0,1)))(input_layer)
        conv = keras.layers.Conv2D(filters=nf, kernel_size=(3,3), strides=(2,2), padding="valid")
        output = keras.layers.concatenate([conv(pad1), conv(pad2)], axis=3)
        return output

We think the issue come from the [delete_node_nto1](https://github.com/microsoft/onnxconverter-common/blob/master/onnxconverter_common/optimizer.py#L392). We look in the function but with the complexity of the function we are not sure to be able to fix it without breaking it.

The code to recreate the issue can be found here : 
https://gist.github.com/edmBernard/0f899f76c5b234ec0189ce327a423900

**Comment by jiafatom** [comment]
Just made a fix https://github.com/microsoft/onnxconverter-common/pull/178
and it looks good now. Please pull onnxconverter-common master and retry. Thanks.

**Comment by edmBernard** [comment]
Thanks a lot. I just check on my side and it work perfectly.
